core/App/ehsm_napi.cpp

```diff
@@ -1125,4 +1129,118 @@ char* NAPI_ExportDataKey(const char* cmk_base64,
     return retJsonObj.toChar();
 }
 
-}  // extern "C"
+/*
+@return
+[string] json string
+    {
+        code: int,
+        message: string,
+        result: {
+            challenge_base64 : string,
+            ga_base64 : string
+        }
+    }
+*/
+char *NAPI_RA_HANDSHAKE_MSG0(const char *p_msg0)
+{
+    print_d("***NAPI_RA_HANDSHAKE_MSG0 start.");
+
+    print_d("msg0: \n %s", p_msg0);
+    std::string challenge_base64 = "challenge_base64";
+    std::string ga_base64 = "ga_base64";
+
+    RetJsonObj retJsonObj;
+    
+    retJsonObj.addData("challenge_base64", challenge_base64);
+    retJsonObj.addData("ga_base64", ga_base64);
+    print_d("msg1: \n%s",retJsonObj.toChar());
+
+    print_d("***NAPI_RA_HANDSHAKE_MSG0 end.");
+    return retJsonObj.toChar();
+}
+
+/*
+@return
+[string] json string
+    {
+        code: int,
+        message: string,
+        result: {
+            msg3_base64 : string
+        }
+    }
+*/
+char *NAPI_RA_HANDSHAKE_MSG2(const char *p_msg2)
+{
+    print_d("***NAPI_RA_HANDSHAKE_MSG2 start.");
+
+    print_d("msg2: \n %s", p_msg2);
+    std::string msg3_base64 = "msg3_base64";
+
+    RetJsonObj retJsonObj;
+    retJsonObj.addData("msg3_base64", msg3_base64);
+    print_d("msg3: \n%s",retJsonObj.toChar());
+
+    print_d("***NAPI_RA_HANDSHAKE_MSG2 end.");
+    return retJsonObj.toChar();
+}
+
+/*
+@return
+[string] json string
+    {
+        code: int,
+        message: string,
+        result: {
+            appid : string
+            apikey : string
+        }
+    }
+*/
+char *NAPI_RA_GET_API_KEY(const char *p_msg4)
+{
+    print_d("***NAPI_RA_GET_API_KEY start.");
+    ehsm_status_t ret = EH_OK;
+    RetJsonObj retJsonObj;
+    std::string appid;
+    std::string apikey;
+
+    print_d("msg4: \n %s", p_msg4);
+
+    // create appid
+    uuid_t uu;
+    uuid_generate(uu);
+    char uuid_str[37];
```

Comment:
where does 37 come from?  sizeof(uuid_t)?

Reply:
uuid length is 36